src/components/TextEditor/components/TextEditor.js

```diff
@@ -199,136 +215,135 @@ class TextEditor extends React.Component {
    * @returns {string} the codemirror theme - see https://codemirror.net/demo/theme.html for more info
    */
 
-  getCMTheme = (theme) => {
-    switch (theme) {
-      case 'light':
-        return 'duotone-light';
-      case 'dark':
-      default:
-        return 'material';
+  const getCMTheme = (theme2) => {
+    switch (theme2) {
+    case 'light':
+      return 'duotone-light';
+    case 'dark':
+    default:
+      return 'material';
     }
   };
 
-  renderDropdown = () => <DropdownButtonContainer />;
+  const renderDropdown = () => <DropdownButtonContainer />;
 
-  renderSketchName = () => <div className="program-sketch-name">{this.props.sketchName}</div>;
-
-  renderBanner = () => {
-    const thumbnail =
-      ThumbnailArray[this.props.viewOnly ? this.props.vthumbnail : this.props.thumbnail];
+  const renderSketchName = () => <div className="program-sketch-name">{sketchName}</div>;
+  const awareButtonReturn = () => {
+    if (viewOnly) {
+      if (uid) {
+        <ViewportAwareButton
+          size="lg"
+          onClick={openForkModal}
+          isSmall={screenWidth <= EDITOR_WIDTH_BREAKPOINT}
+          icon={<FontAwesomeIcon icon={faCodeBranch} />}
+          text="Fork"
+        />;
+      }
+    } else {
+      <ViewportAwareButton
+        className="mx-2"
+        color="success"
+        size="lg"
+        onClick={handleSave}
+        isSmall={screenWidth <= EDITOR_WIDTH_BREAKPOINT}
+        icon={<FontAwesomeIcon icon={faSave} />}
+        text={saveText}
+      />;
+    }
+  };
+  const renderBanner = () => {
+    const athumbnail = ThumbnailArray[viewOnly ? vthumbnail : thumbnail];
     return (
       <div className="code-section-banner">
         <OpenPanelButtonContainer />
         <img
           className="program-sketch-thumbnail"
-          src={`${process.env.PUBLIC_URL}/img/sketch-thumbnails/${thumbnail}.svg`}
+          src={`${process.env.PUBLIC_URL}/img/sketch-thumbnails/${athumbnail}.svg`}
           alt="sketch thumbnail"
         />
-        {this.props.viewOnly ? this.renderSketchName() : this.renderDropdown()}
+        {viewOnly ? renderSketchName() : renderDropdown()}
```

Comment:
I don't see a reason why the component can't just directly be placed here (`<DropdownButtonContainer />`)